src/nvim/strings.c

```diff
@@ -45,6 +45,41 @@
 #include "nvim/os/os.h"
 #include "nvim/os/shell.h"
 
+/// Compares two strings for equality.
+bool strequal(const char *a, const char *b)
+  FUNC_ATTR_NONNULL_ALL FUNC_ATTR_PURE
+{
+  return strcmp(a, b) == 0;
+}
+
+/// Compares two sized strings for equality.
+bool strnequal(const char *a, const char *b, size_t size)
+  FUNC_ATTR_NONNULL_ALL FUNC_ATTR_PURE
```

Comment:
No: http://www.nongnu.org/c-prog-book/online/x1019.html

In effect, dereferencing the pointer arguments `a` and `b` (which `strcmp()` does by necessity) is equal to "reading global memory" from gcc's point of view. `const` functions are thus only for the case where no pointers are dereferenced (though it doesn't hurt to just read the pointer values themselves).